src/gsy_e/models/strategy/future/strategy.py

```diff
@@ -0,0 +1,258 @@
+from gsy_e.models.base import AssetType
+from gsy_e.models.strategy.update_frequency import (
+    TemplateStrategyBidUpdater, TemplateStrategyOfferUpdater, TemplateStrategyUpdaterInterface)
+
+if TYPE_CHECKING:
+    from gsy_e.models.area import Area
+    from gsy_e.models.strategy import BidEnabledStrategy
+    from gsy_e.models.market.future import FutureMarkets
+
+
+class FutureTemplateStrategyBidUpdater(TemplateStrategyBidUpdater):
+    """Version of TemplateStrategyBidUpdater class for future markets"""
+
+    @property
+    def _time_slot_duration_in_seconds(self) -> int:
+        return GlobalConfig.FUTURE_MARKET_DURATION_HOURS * 60 * 60
+
+    @staticmethod
+    def get_all_markets(area: "Area") -> List["FutureMarkets"]:
+        """Override to return list of future markets"""
+        return [area.future_markets]
+
+    @staticmethod
+    def get_all_time_slots(area: "Area") -> List[DateTime]:
+        """Override to return all future market available time slots"""
+        return area.future_markets.market_time_slots
+
+    def update(self, market: "FutureMarkets", strategy: "BidEnabledStrategy") -> None:
+        """Update the price of existing bids to reflect the new rates."""
+        for time_slot in strategy.area.future_markets.market_time_slots:
+            if self.time_for_price_update(strategy, time_slot):
+                if strategy.are_bids_posted(market.id, time_slot):
+                    strategy.update_bid_rates(market, self.get_updated_rate(time_slot))
+
+
+class FutureTemplateStrategyOfferUpdater(TemplateStrategyOfferUpdater):
+    """Version of TemplateStrategyOfferUpdater class for future markets"""
+
+    @property
+    def _time_slot_duration_in_seconds(self) -> int:
+        return GlobalConfig.FUTURE_MARKET_DURATION_HOURS * 60 * 60
+
+    @staticmethod
+    def get_all_markets(area: "Area") -> List["FutureMarkets"]:
+        """Override to return list of future markets"""
+        return [area.future_markets]
+
+    @staticmethod
+    def get_all_time_slots(area: "Area") -> List[DateTime]:
+        """Override to return all future market available time slots"""
+        return area.future_markets.market_time_slots
+
+    def update(self, market: "FutureMarkets", strategy: "BidEnabledStrategy") -> None:
+        """Update the price of existing offers to reflect the new rates."""
+        for time_slot in strategy.area.future_markets.market_time_slots:
+            if self.time_for_price_update(strategy, time_slot):
+                if strategy.are_offers_posted(market.id):
+                    strategy.update_offer_rates(market, self.get_updated_rate(time_slot))
+
+
+class FutureMarketStrategyInterface:
+    """Dummy/empty class that does not provide concrete implementation of the methods.
+    Is needed in order to disable the implementation of the future market strategy
+    when future markets are disabled by configuration."""
+    def __init__(self, *args, **kwargs):
+        pass
```

Comment:
Is this really needed?

Reply:
I think it is a remnant of the implementation, will give it a try without it, thanks!